### ISSUE TITLE
resp.once is not a function error thrown

### ISSUE THREAD

**Issue by jrcastillo** [ziyasal/hapijs-status-monitor#4]
This error occurs after starting the server and request the route `/status`. The stack trace is the following:
```
Debug: internal, implementation, error 
    TypeError: Uncaught error: resp.once is not a function
    at /home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapijs-status-monitor/index.js:112:10
    at /home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/request.js:402:22
    at iterate (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at Object.exports.serial (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/node_modules/items/lib/index.js:39:9)
    at /home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/request.js:397:15
    at internals.Protect.run.finish [as run] (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/protect.js:64:5)
    at internals.Request.internals.Request._execute.internals.Request._lifecycle.internals.Request._invoke.callback._protect.run [as _invoke] (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/request.js:395:19)
    at internals.Request.internals.Request._execute.internals.Request._lifecycle.internals.Request._invoke.internals.Request._reply.exit [as _reply] (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/request.js:454:10)
    at /home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/request.js:388:21
    at done (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/node_modules/items/lib/index.js:31:25)
    at finalize (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/handler.js:21:16)
    at /home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/protect.js:52:16
    at wrapped (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at finalize (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/handler.js:82:20)
    at Function.wrapped [as _next] (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at Function.internals.Reply.interface.internals.response (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/reply.js:137:14)
    at reply (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/reply.js:70:22)
    at internals.Connection._init.internals.Connection._start.internals.Connection._stop.internals.Connection._dispatch.internals.Connection.inject.internals.notFound (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/connection.js:397:12)
    at Object.exports.execute.internals.prerequisites.internals.handler.callback [as handler] (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/handler.js:96:36)
    at /home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/handler.js:30:23
    at internals.Protect.run.finish [as run] (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/protect.js:64:5)
    at exports.execute.finalize (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/handler.js:24:22)
    at each (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/request.js:383:16)
    at iterate (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at Object.exports.serial (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/node_modules/items/lib/index.js:39:9)
    at internals.Request.internals.Request._execute.internals.Request._lifecycle.each [as _lifecycle] (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/request.js:386:11)
    at internals.Request.internals.Request._execute (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/request.js:301:21)
    at Domain.<anonymous> (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/connection.js:244:25)
    at Domain.run (domain.js:228:14)
    at internals.Protect.run.internals.Protect.enter (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/protect.js:80:17)
    at Server.<anonymous> (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapi/lib/connection.js:242:30)
    at Server.<anonymous> (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapijs-status-monitor/node_modules/socket.io/node_modules/engine.io/lib/server.js:435:22)
    at Server.<anonymous> (/home/ramon/Desktop/ZOI/ZOI3/zoi-api/node_modules/hapijs-status-monitor/node_modules/socket.io/lib/index.js:260:16)
    at emitTwo (events.js:87:13)
    at Server.emit (events.js:172:7)
    at HTTPParser.parserOnIncoming [as onIncoming] (_http_server.js:528:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:103:23)
```

**Comment by Sniperkillerut** [comment]
I have the same issue, but it only happens on a jwt secured page on a 401 error

**Comment by jrcastillo** [comment]
It's weird, for me it happens for the route given out in the documentation `/status` and does not record any type of metrics or profiling. Can't figure out whats happening. I'm using `jwt` for route authentication but since `/status` is used internally for this plugin, there is no explicit implementation of `jwt`

**Comment by Sniperkillerut** [comment]
This fixed it, just added: 
       if (!request.response.isBoom) { 
to:
\node_modules\hapijs-status-monitor\index.js
line 106
  /*  hook into the middle of processing  */
  server.ext('onPreResponse', (request, reply) => {
    if (!request.response.isBoom) {
      const resp = request.response;
      let statusCode = resp.statusCode;
      const startTime = process.hrtime();

      resp.once('finish', () => {
        const diff = process.hrtime(startTime);
        const responseTime = diff[0] * 1e3 + diff[1] * 1e-6;
        const category = Math.floor(statusCode / 100);

        options.spans.forEach((span) => {
          const lastItem = span.responses[span.responses.length - 1];
          if (lastItem !== undefined &&
            lastItem.timestamp / 1000 + span.interval > Date.now() / 1000) {
            lastItem[category]++;
            lastItem.count++;
            lastItem.mean = lastItem.mean + ((responseTime - lastItem.mean) / lastItem.count);
          } else {
            span.responses.push({
              '2': category === 2 ? 1 : 0,
              '3': category === 3 ? 1 : 0,
              '4': category === 4 ? 1 : 0,
              '5': category === 5 ? 1 : 0,
              count: 1,
              mean: responseTime,
              timestamp: Date.now()
            });
          }
        });
      });

    }
      return reply.continue()
  })

  /*  continue processing  */
  return next()
}

**Comment by jrcastillo** [comment]
I tried out your solution and got then a missing authentication like the route was defaulted to ask for `jwt`, I generated a `jwt` just for the request and got again an error. But turns out this has to do with some configurations at my `server.js`. 

It seems I'll have to manage or find out where in my code is the response being modified and making the plugin crash. Thanks, the monitor works beautifully

**Comment by Sniperkillerut** [comment]
well it happens on a onPreResponse function when the response is empty (usually when there is an error) so it needs fixing somehow, either a
 if (!request.response.isBoom) {
 if (!request.response) { //not null
or something, keep the issue open until a real fix is made by a contributor

**Comment by danielbayerlein** [comment]
@ziyasal This bug is resolved with #7.. You can close this issue. :smiley:

**Comment by ziyasal** [comment]
hey @danielbayerlein thanks!
